addons/l10n_in/models/account_invoice.py

```diff
@@ -148,3 +149,23 @@ def _l10n_in_get_hsn_summary_table(self):
                 'taxes_data': taxes_data,
             })
         return self.env['account.tax']._l10n_in_get_hsn_summary_table(base_lines, display_uom)
+
+    @api.depends('invoice_line_ids.tax_ids', 'l10n_in_gst_treatment', 'l10n_in_state_id.code')
+    def _compute_l10n_in_warnings(self):
+        for move in self:
+            if move.country_code != 'IN':
+                move.l10n_in_warnings = {}
+                continue
+            warnings = {}
+            if move.l10n_in_state_id.code == 'IN_OC' and move.l10n_in_gst_treatment == 'overseas':
+                if any(
+                    not self.env.ref('l10n_in.tax_tag_igst') in tax.invoice_repartition_line_ids.tag_ids
+                    for tax in move.invoice_line_ids.tax_ids.filtered(lambda tax: tax.country_code == 'IN')
```

Comment:
Multiple things to say here:
Avoid doing a double loop here (your list comprehension and the filtered function).
Instead do something like this:

```python
x = [
    a
    for a in my_list
    if a.country_code == 'IN'
]
```

After, avoid using `.ref()` in a loop, each time, you'll `browse` and call `exists` on the same new object. Save it in a variable before entering in your list comprehension.

Finally, instead of doing the` not self.env.ref('l10n_in.tax_tag_igst') in tax.invoice_repartition_line_ids.tag_ids`, inverse it for something like this: 

```python
self.env.ref('l10n_in.tax_tag_igst') not in tax.invoice_repartition_line_ids.tag_ids
```

Reply:
I have updated the code according to the last-minute feedback from the product owner for improved functionality. I will incorporate the changes you mentioned and push the new code soon.  @FlorianGilbert 
 
 Thank you